motioneye/uploadservices.py

```diff
@@ -881,6 +882,79 @@ def _refresh_credentials(self, refresh_token):
         }
 
 
+class Webdav(UploadService):
+    NAME = 'webdav'
+
+    def __init__(self, camera_id):
+        self._server = None
+        self._port = None
+        self._username = None
+        self._password = None
+        self._location = None
+
+        UploadService.__init__(self, camera_id)
+
+    def _request(self, url, method, body=None):
+        base64string = b64encode(f'{self._username}:{self._password}'.encode()).decode('ascii')
+        headers = {'Authorization': f'Basic {base64string}'}
+        if body is not None:
+            headers.update('Content-Length', len(body))
+        self.debug(f'request: {method} {url}')
+        request = urllib.request.Request(url, data=body, headers=headers)
+        request.get_method = lambda: method
+        try:
+            utils.urlopen(request)
+        except urllib.HTTPError as e:
+            if method == 'MKCOL' and e.code == 405:
+                self.debug(
+                    'MKCOL failed with code 405, this is normal if the folder exists'
+                )
+            else:
+                raise e
+
+    def _make_dirs(self, path):
+        dir_url = self._server
+        for folder in path.strip('/').split('/'):
+            dir_url = dir_url + folder + '/'
```

Comment:
Similar issue than with below missing slashes: Here the first "folder" is added to the hostname without slash separation:
```suggestion
        dir_url = self._server + '/'
        for folder in path.strip('/').split('/'):
            dir_url = dir_url + folder + '/'
```